health/health.c

```diff
@@ -323,7 +340,13 @@ static inline void health_alarm_execute(RRDHOST *host, ALARM_ENTRY *ae) {
     char buffer[100 + 1];
     while(fgets(buffer, 100, fp) != NULL) ;
     ae->exec_code = mypclose(fp, command_pid);
-    debug(D_HEALTH, "done executing command - returned with code %d", ae->exec_code);
+    gettimeofday(&command_end_time, NULL);
+    unsigned long command_total_time =
```

Comment:
I confess this math is not clear for me, you are advancing  almost 12 days tv_sec  and after this you sum the microseconds value, I suspect you wanted to convert the microseconds into seconds to do the sum. 
I will give a different suggestion for you, instead we use only one variable to store the total time, we could do two subtractions and store the result as SECONDS.MICROSECONDS. For more details you can see this example https://www.gnu.org/software/libc/manual/html_node/Elapsed-Time.html .

Reply:
I'm converting the sec into µsec (* 1,000,000), then adding the usec to the result. Then subtracting both µsec times and converting it to a float in sec ( / 1,000,000)
While I usually avoid floats, we already are in the flow of creating a process and in a non time-sensitive thread. And the optimizer has a lot of trick with the optimization: [Look at code diff with Compiler Explorer](https://gcc.godbolt.org/z/ddHgkM).
Also note that [gettimeofday is a VDSO](https://github.com/torvalds/linux/blob/3e89bf35ebf59c12e8c1476f6681fae0ebdcb2a7/arch/x86/entry/vdso/vclock_gettime.c#L204-L218) so has a low overhead. I would have used `clock_gettime(CLOCK_REALTIME_COARSE)` but it's too linux specific.

Reply:
Ok, now I understood what you want to do, I had a confusion because the field stores seconds since 01/01/1970.
Thank you, you are doing a good job here! :)